modules/nextflow/src/main/groovy/nextflow/cloud/aws/AmazonCloudDriver.groovy

```diff
@@ -168,7 +182,16 @@ class AmazonCloudDriver implements CloudDriver {
      */
     protected String fetchIamRole() {
         try {
-            def role = getUrl('http://169.254.169.254/latest/meta-data/iam/security-credentials/').readLines()
+	    final urlbase = getUrlBase()
+	    def url = ""
+	    
+	    if( inEcs() ) {
+		final env = System.getenv()
+		url = urlbase + env['AWS_CONTAINER_CREDENTIALS_RELATIVE_URI']
+	    }
+	    else
+		url = urlbase + '/latest/meta-data/iam/security-credentials/'
+            def role = getUrl(url).readLines()
             if( role.size() != 1 )
```

Comment:
Thanks for this PR. IMO it would make more sense to add a `getSecurityCredentialsUrl` which checks the sys env and return the complete path (and maybe rename `getUrl` to `readUrl`)